### PR TITLE
Implement basic loading component

### PR DESCRIPTION
**Pull request by jeremiak**
Changes the stores to have a `.fetching` attribute, which will be a boolean. How the stores implement the toggling of this value is up to each individual store. Then the components can check for requests that are currently in flight and use the `<Loading />` component to display a standardized loading message.

We still need to design the loading component, as it is pretty stark right now but it is a first pass.

Refs #383

### CODE REVIEW

**Comment by jeremiak** [comment]
This is ready for review @msecret

**Comment by jcscottiii** [comment]
@jeremiak This might be more for another PR / I'm missing something but I'm wondering does this account for a certain model type (e.g. users) doing multiple fetching at the same time on the page. I'm sure it doesn't happen now but just in case it happens in the future.

**Comment by jeremiak** [comment]
@jcscottiii it is a great point. Just to clarify, you're saying that if:
* Request A is made and `fetching` is set to true
* Request B is made and `fetching` is set to true, which means it doesn’t change
* Request A comes back and changes fetching to false and therefore the store is `fetching` false but there is still a request in flight

Therefore, the store is actually waiting for a request but it's `fetching` value doesn't reflect that. This is definitely a problem with this design/implementation. I was trying to keep the mechanism really simple and made that tradeoff. I haven't seen it be a problem with any of the stores yet, but it could just be that my network connection has worked.

Even if we were to keep track of each request going out, it would still be hard to know if every request an interface needed had come back with fresh data or if it was stale. Thoughts on how we might want to do this otherwise?

**Comment by jcscottiii** [comment]
let me think of it this way:
```
# single store emits requests all while rendering a single fetch 
requests (1..N) <- single store (contains) single fetching <-> render single loading state
```

instead let's do it this way make the fetching as part of the component instead of the store. thoughts @msecret @jeremiak ?

**Comment by msecret** [comment]
According to our design, only the store can keep state, such as fetching. But the components can be listening to the store.

**Comment by jcscottiii** [comment]
@msecret ahh yeah. hmmm. so could each component get a way to know that the store is changing it's particular fetching state?

**Comment by msecret** [comment]
Yea I mean, we coudl keep fetching in the stores but separate different fetching out more so each fetching is more specific? Then components could be more specific about what fetching they're waiting for?

**Comment by jcscottiii** [comment]
@msecret i got ya!

**Comment by msecret** [comment]
If this works, we should use it for the time being, and get these merge conflicts fixed. @jcscottiii if you're interested in taking a look at improving the fetching tracking in the next few days as we just discussed, you definitely should, I think it could be a good improvement.

**Comment by jeremiak** [comment]
@msecret @jcscottiii I fixed the merge conflicts so this should be ready for review now

File: static_src/components/loading.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+
+import React from 'react';
+
+const propTypes = {
```
Comment (msecret) [inline static_src/components/loading.jsx]: Is there a reason you defined these into separate variables rather then directly like other components:

```
Loading.proptTypes = {
  text
  ...
```

Is this a better pattern we should adopt?

**Pull request merged**